src/shapes/tessellation_twist/TessellationTwist.js

```diff
@@ -0,0 +1,266 @@
+  vec2( 0.0,  0.9814954573),
+]
+
+const vec_square = [
+  vec2(-0.7, -0.7),
+  vec2( 0.7,  0.7),
+  vec2(-0.7,  0.7),
+
+  vec2(-0.7, -0.7),
+  vec2( 0.7,  0.7),
+  vec2( 0.7, -0.7)
+]
+
+var vec_pentagon = [
+  vec2(0.0, 0.0),
+  vec2(0.0, -1.000000),
+  vec2(-0.951057, -0.309017),
+
+  vec2(0.0, 0.0),
+  vec2(-0.951057, -0.309017),
+  vec2(-0.587785, 0.809017),
+
+  vec2(0.0, 0.0),
+  vec2(-0.587785, 0.809017),
+  vec2(0.587785, 0.809017),
+
+  vec2(0.0, 0.0),
+  vec2(0.587785, 0.809017),
+  vec2(0.951057, -0.309017),
+
+  vec2(0.0, 0.0),
+  vec2(0.951057, -0.309017),
+  vec2(0.0, -1.000000),
+]
+
+var vec_hexagon = [
+  vec2(0.0, 0.0),
+  vec2(0.0, -1.000000),
+  vec2(-0.866025, -0.500000),
+
+  vec2(0.0, 0.0),
+  vec2(-0.866025, -0.500000),
+  vec2(-0.866025, 0.500000),
+
+  vec2(0.0, 0.0),
+  vec2(-0.866025, 0.500000),
+  vec2(0.0, 1.000000),
+
+  vec2(0.0, 0.0),
+  vec2(0.0, 1.000000),
+  vec2(0.866025, 0.500000),
+
+  vec2(0.0, 0.0),
+  vec2(0.866025, 0.500000),
+  vec2(0.866025, -0.500000),
+
+  vec2(0.0, 0.0),
+  vec2(0.866025, -0.500000),
+  vec2(0.0, -1.000000),
+]
+
+function getEdges(edges, a, b, c, count, settings) {
+  let da, db, dc
+
+  if (count === 0) {
+    if (settings.rotate > 0) {
+      da = Math.sqrt(Math.pow(a[0], 2) + Math.pow(a[1], 2)) * (settings.rotate * Math.PI / 180.0)
+      db = Math.sqrt(Math.pow(b[0], 2) + Math.pow(b[1], 2)) * (settings.rotate * Math.PI / 180.0)
+      dc = Math.sqrt(Math.pow(c[0], 2) + Math.pow(c[1], 2)) * (settings.rotate * Math.PI / 180.0)
+    } else {
+      da = (settings.rotate * Math.PI / 180.0)
+      db = (settings.rotate * Math.PI / 180.0)
+      dc = (settings.rotate * Math.PI / 180.0)
+    }
+
+    let ap = vec2(
+      (a[0] * Math.cos(da)) - (a[1] * Math.sin(da)),
+      (a[0] * Math.sin(da)) + (a[1] * Math.cos(da)))
+    let bp = vec2(
+      (b[0] * Math.cos(db)) - (b[1] * Math.sin(db)),
+      (b[0] * Math.sin(db)) + (b[1] * Math.cos(db)))
+    let cp = vec2(
+      (c[0] * Math.cos(dc)) - (c[1] * Math.sin(dc)),
+      (c[0] * Math.sin(dc)) + (c[1] * Math.cos(dc)))
+
+    edges.push([ap, bp], [ap, cp], [bp, cp])
+    return
+  }
+
+  let ab = mix(a, b, 0.5)
+  let ac = mix(a, c, 0.5)
+  let bc = mix(b, c, 0.5)
+
+  getEdges(edges, ab, ac, bc, count - 1, settings)
+  getEdges(edges, c, ac, bc, count - 1, settings)
+  getEdges(edges, b, bc, ab, count - 1, settings)
+  getEdges(edges, a, ab, ac, count - 1, settings)
+}
+
+const options = {
+  ...shapeOptions,
+  ...{
+    tessellationTwistNumSides: {
+      title: "Number of sides",
+      min: 3,
+      max: 6
+    },
+    tessellationTwistIterations: {
+      title: "Iterations",
+      min: 0,
+      max: 4
+    },
+    tessellationTwistRotate: {
+      title: "Rotate and twist",
+      step: 5,
+      min: 0
+    }
+  }
+}
+
+// Adapted from https://codepen.io/rafaelpascoalrodrigues/pen/KpBJve.
+export default class TessellationTwist extends Shape {
+  constructor() {
+    super('Tessellation Twist')
+  }
```

Comment:
Is there a link for this one? It looks like there is a lot of interesting math, including some travelling salesman solving?